proto/vschema.proto

```diff
@@ -45,6 +45,15 @@ message Keyspace {
   map<string, Table> tables = 3;
   // If require_explicit_routing is true, vindexes and tables are not added to global routing
   bool require_explicit_routing = 4;
+  // foreign_key_mode dictates how Vitess should handle foreign keys for this keyspace.
+  ForeignKeyMode foreign_key_mode = 5;
+
+  enum ForeignKeyMode {
+    Fk_default = 0;
```

Comment:
What's the idea of `default`? People will automatically get new behavior when we change the default behavior? I don't think this is such a good idea. Opting in and out should be explicit and new users who don't want to choose can get whatever the default is in that release.